### ISSUE TITLE
CLI: User answered "No" - do not trigger respective hook.

### ISSUE THREAD

**Issue by b12k** [release-it/release-it#635]
Hello.
Thanx for a awesome tool!

There is one thing:
When using cli and going from step to step user in prompted if to trigger certain action or not.
In our configuration we are using hooks to perform extra actions after certain step.
This hooks are triggered even if user choosed not to perform certain action.

```
? Create a release on GitHub (Release 0.0.6)? No
... after:gihtub:relese actions triggered even if answer is "No"

... before:npm:relese actions triggered even if its not necessary
? Publish @username/repository to npm? No
```
![screenshot](https://i.imgur.com/Rr5XDnh.png)

I would expect that prompt will happen before triggering hooks and will decide if execution should happen.

Thanx in advance!

**Comment by webpro** [comment]
That's a fair point, thanks for reporting this. I'll look into it soon.

**Comment by webpro** [comment]
In the plugin architecture of release-it, the hooks (configuration) are completely separated from the plugin lifecycle commands (which might show or not show prompts). Put differently, plugins don't report back to the task runner. This is why you encounter the behavior you describe.

The most I could do at this point is to not run `after:*:*` hooks when either the user answers this prompt with "No", or when it is configured not to run. The `before:*:*` will still always run (otherwise, the coupling/logic gets awkward and it would be a breaking change for external plugins). I'm leaning towards thinking this is is an acceptable solution.

**Comment by webpro** [comment]
I have just released [v13.5.4](https://github.com/release-it/release-it/releases/tag/13.5.4), which results in not executing `after:*:release` hooks when they are disabled or skipped (or any `after:*:*`, to be more precise). Note that the `before:*:release` hooks are still always executed.

Let me know how it works for you and whether you have any questions.
